source/material_model/utilities.cc

```diff
@@ -472,28 +472,78 @@ namespace aspect
           // Read data from disk and distribute among processes
           std::istringstream in(Utilities::read_and_distribute_file_content(filename, comm));
 
-          std::getline(in, temp); // eat first line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-
-          in >> min_temp;
-          std::getline(in, temp);
-          in >> delta_temp;
-          std::getline(in, temp);
-          in >> n_temperature;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          in >> min_press;
-          min_press *= 1e5;  // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> delta_press;
-          delta_press *= 1e5; // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> n_pressure;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          std::getline(in, temp);
+          // The following lines read in a PerpleX tab file in standard format
+          // First line is the Perplex version number
+          std::getline(in, temp); // get next line, table file name
+          std::getline(in, temp); // get next line, dimension of table
+          std::getline(in, temp); // get next line, either T(K) or P(bar)
+
+          for (unsigned int i=0; i<2; i++)
+            {
+              std::string natural_variable;
+              in >> natural_variable;
+
+              if (natural_variable[0] == 'T')
```

Comment:
I think it would greatly help to understand the order of variables if you could copy-paste a typical file header here. This way, we would see which lines you're ignoring in the various `std::getline(in, temp);` below.

Reply:
Done.